app/scripts/controllers/transactions/pending-tx-tracker.js

```diff
@@ -214,6 +224,9 @@ class PendingTransactionTracker extends EventEmitter {
     const address = txMeta.txParams.from
     const completed = this.getCompletedTransactions(address)
     const sameNonce = completed.filter((otherMeta) => {
+      if (otherMeta.id === txMeta.id) {
```

Comment:
Ohh this is tricky. I'm guessing this is here because the transaction being checked may have completed since `_checkPendingTx` was called? Unless it's possible for two txs to have the same id somehow.

Reply:
yes it is because of the possibility that as this is running the tx has completed

Reply:
so if we dont check the id but we get it in the list we will mark it as dropped with out this line